### PR TITLE
Tagging

### LINKED ISSUE #34: Tags for Section Sub Categories

### PR DESCRIPTION
**Pull request by EricLarson2020**
### Description
Added tags to the phrase model

### Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Refactor (non-breaking change which maintains existing functionality)
 - [yes] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: app/graphql/types/phrase_type.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@ class PhraseType < Types::BaseObject
     field :id, ID, null: false
     field :expression, String, null: false
     field :image, String, null: false
```
Comment (EricLarson2020) [inline app/graphql/types/phrase_type.rb:5]: added tags to phrases

File: db/migrate/20200909191545_add_tags_to_phrases.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddTagsToPhrases < ActiveRecord::Migration[5.2]
+  def change
+    add_column :phrases, :tags, :string, array: true, default: []
```
Comment (EricLarson2020) [inline db/migrate/20200909191545_add_tags_to_phrases.rb:3]: added in column of tags as an array of phrases

File: db/seeds.rb [hunk 3]
```diff
@@ -11,5 +11,5 @@
 user_1 = User.create(firstName: "Bob", lastName: "Vance", email: "bob@email.com", password:"123", speed:1, voice: "Matt")
 UserSection.create(user_id: user_1.id, section_id: section_1.id)
 UserSection.create(user_id: user_1.id, section_id: section_2.id)
-phrase_1 = Phrase.create(expression: "Hello", image: "fake.url")
+phrase_1 = Phrase.create(expression: "Hello", image: "fake.url", tags: ['greeting', 'hello'])
```
Comment (EricLarson2020) [inline db/seeds.rb:14]: updated seeds

File: spec/requests/v1/phrase_query_spec.rb [hunk 4]
```diff
@@ -1,29 +1,31 @@
...
         post "/v1/graphql", params: { query: query(id: hello.id)}
-        expect(response).to be_successful  
+        expect(response).to be_successful
 
         reply = JSON.parse(response.body, symbolize_names: true)
         expect(reply[:data][:phrase][:expression]).to_not be_empty
         expect(reply[:data][:phrase][:image]).to_not be_empty
-
-    end    
+        expect(reply[:data][:phrase][:tags]).to eq(['hello', 'goodbye'])
```
Comment (EricLarson2020) [inline spec/requests/v1/phrase_query_spec.rb:17]: updated test to include new tags

File: Gemfile [hunk 5]
```diff
@@ -34,6 +34,7 @@ group :development, :test do
   gem 'factory_bot_rails'
   gem 'faker'
   gem 'graphiql-rails'
+  gem 'simplecov'
```
Comment (sagemlee) [inline Gemfile:37]: Thanks for adding this in.

**Review by sagemlee — APPROVED** [review]

File: db/seeds.rb (hunk 3 above)
Comment (cgaddis36) [reply db/seeds.rb:14]: love the tags

File: Gemfile (hunk 5 above)
Comment (cgaddis36) [reply Gemfile:37]: yes, super nice

File: app/graphql/types/phrase_type.rb [hunk 6]
```diff
@@ -3,7 +3,8 @@ class PhraseType < Types::BaseObject
...
+    field :tags, [String], null: false
```
Comment (cgaddis36) [inline app/graphql/types/phrase_type.rb:6]: this is interesting, with the square brackets around String. Nice find

File: spec/requests/v1/phrase_query_spec.rb (hunk 4 above)
Comment (cgaddis36) [reply spec/requests/v1/phrase_query_spec.rb:17]: nice testing updates

**Review by cgaddis36 — APPROVED** [review]

**Pull request merged**